src/cmd-import

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/python3
+
+import argparse
+import datetime
+import json
+import os
+import subprocess
+import tempfile
+import shutil
+from stat import (
+    S_IREAD,
+    S_IRGRP,
+    S_IROTH)
+from cosalib.builds import Builds
+from cosalib.cmdlib import (
+    rfc3339_time,
+    get_basearch,
+    sha256sum_file)
+
+VALID_NAMES = ["fedora-coreos", "rhcos", "scos"]
+
+
+def main():
+    args = parse_args()
+    with tempfile.TemporaryDirectory(prefix='cosa-import-', dir='tmp') as tmpd:
+        tmp_ociarchive = os.path.join(tmpd, "out.ociarchive")
+
+        metadata = prepare_build(args, tmp_ociarchive)
+
+        name = metadata['Labels']['com.coreos.osname']
+        buildid = metadata['Labels']['org.opencontainers.image.version']
+
+        builds = Builds()
+        if builds.has(buildid):
+            raise Exception(f"Build ID {buildid} already exists!")
```

Comment:
We can call skopeo inspect on the srcimg to get the metadata and avoid downloading the archive if build already exists.